hexl/ntt/ntt-radix-2.cpp

```diff
@@ -203,10 +287,19 @@ void InverseTransformFromBitReverseRadix2(
           const uint64_t W = inv_root_of_unity_powers[root_index];
           const uint64_t W_precon = precon_inv_root_of_unity_powers[root_index];
 
-          uint64_t* X = operand + j1;
-          uint64_t* Y = X + t;
-
-          InvButterflyRadix2(X, Y, W, W_precon, modulus, twice_modulus);
+          if(result != operand){
+            uint64_t* X_r = result + j1;
+            uint64_t* Y_r = X_r + t;
+            const uint64_t* X_op = operand + j1;
+            const uint64_t* Y_op = X_op + t;
+            InvButterflyRadix2(X_r, Y_r, X_op, Y_op, W, W_precon,
```

Comment:
Can we just have a single out-of-place InvButterflyRadix2 and pass the same arguments as input/output?

Reply:
In this case you want me try only having the out-of-place for this first pass or everywhere?

Reply:
In general, if performance isn't degraded, I'd prefer having just one (out-of-place) impl. for each function, rather than two separate impls. So, "everywhere" :-)

Reply:
Ok, I did the change just on the first pass and attached the results up in the first comment.
I can put it here also. [Performance.xlsx](https://github.com/intel/hexl/files/7281621/Performance.xlsx)

Reply:
IIUC, looks like no perf degradation due to this change? That's great.